test/MinimalDelegation.execute.t.sol

```diff
@@ -592,4 +592,113 @@ contract MinimalDelegationExecuteTest is TokenHandler, DelegationHandler, Execut
+        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        bytes memory signature = abi.encode(
+            nonce, abi.encode(webAuthnP256Key.toKeyHash(), webAuthnP256Key.sign(signerAccount.hashTypedData(signedCalls.hash())))
+        );
+        bytes memory executionData1 = abi.encode(calls, signature);
+
+        // create a second batch of calls
+        calls = CallBuilder.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 2e18)); // Transfer 2 tokenA
+
+        // resister a new key to use for the second batch
+        TestKey memory p256Key = TestKeyManager.initDefault(KeyType.P256);
+
+        vm.prank(address(signerAccount));
+        signerAccount.register(p256Key.toKey());
+
+        // Create hash of the calls + nonce and sign it
+        nonce++; // increment the nonce since first one hasnt been used yet
+        signedCalls = SignedCalls({calls: calls, nonce: nonce});
+
+        // Sign using the registered P256 key
+        signature = abi.encode(
+            nonce, abi.encode(p256Key.toKeyHash(), p256Key.sign(signerAccount.hashTypedData(signedCalls.hash())))
+        );
+        bytes memory executionData2 = abi.encode(calls, signature);
+
+
+        bytes[] memory executionDataArray = new bytes[](2);
+        executionDataArray[0] = executionData1;
+        executionDataArray[1] = executionData2;
+
+        // execute the batch of batches
+        signerAccount.execute(BATCH_OF_BATCHES_CALL, abi.encode(executionDataArray));
+        vm.snapshotGasLastCall("execute_BATCH_OF_BATCHES_twoBatches_oneCallEach");
+
+        // Verify the transfers succeeded
+        assertEq(tokenA.balanceOf(address(receiver)), 3e18);
+
+        // Verify the seq was incremented - sequence should increase by 2
+        assertEq(signerAccount.getSeq(0), 2);
+    }
+
+    function test_execute_batchOfBatches_revertsWithCallFailed() public {
+        Call[] memory calls = CallBuilder.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
+
+        // Get the current nonce components for key 0
+        uint256 nonceKey = 0;
+        (uint256 nonce,) = _buildNextValidNonce(nonceKey);
+
+        // Create hash of the calls + nonce and sign it
+        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        bytes32 hashToSign = signerAccount.hashTypedData(signedCalls.hash());
+        bytes memory signature = signerTestKey.sign(hashToSign);
+
+        // Pack the execution data:
+        // 1. Encode the nonce and signature into opData
+        bytes memory opData = abi.encode(nonce, abi.encode(KeyLib.ROOT_KEY_HASH, signature));
+        // 2. Encode the calls and opData together
+        bytes memory executionData1 = abi.encode(calls, opData);
+
+        calls = CallBuilder.init();
+        // this call silently reverts because its over the balance
```

Comment:
this doesn't silently revert though right? it does cause a revert

Reply:
yess forgot to update the comment